core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -249,6 +251,13 @@ public long skip(long n) throws IOException {
       return 0;
     }
 
+    /**
+     * TODO(peis): Figure this out with Calvin or whoever wrote this.
+     * The current implementation looks strange to me. I think it indents to skip n bytes for
+     * caching. I think it should either be implemented the same as seek or read.
```

Comment:
Done except that I am not calling seekInternal inside skip because seek doesn't allow seek(mFileLength) but skip allows.  This is related to the other comment you made about seek.